src/app/dev/DevToys.Tools/Tools/Formatters/Json/JsonFormatterGuiTool.cs

```diff
@@ -0,0 +1,184 @@
+                        Setting("json-text-sortProperties-setting")
+                        .Icon("FluentSystemIcons", '\uf7ed')
+                        .Title(JsonFormatter.SortProperties)
+                        .Description(JsonFormatter.SortPropertiesOptionDescription)
+                        .Handle(
+                            _settingsProvider,
+                            sortProperties,
+                            OnSettingChanged
+                        )
+                    )
+                ),
+                Cell(
+                    GridRow.Content,
+                    GridColumn.Content,
+                    SplitGrid()
+                        .Vertical() 
+                        .WithLeftPaneChild(
+                            _inputTextArea
+                                .Title(JsonFormatter.Input)
+                                .OnTextChanged(OnInputTextChanged))
+                        .WithRightPaneChild(
+                            _outputTextArea
+                                .Title(JsonFormatter.Output)
+                                .ReadOnly()
+                                .Extendable())
+                )
+            )
+        );
+
+    // Smart detection handler.
+    public void OnDataReceived(string dataTypeName, object? parsedData)
+    {
+        if (dataTypeName == PredefinedCommonDataTypeNames.Json &&
+            parsedData is string jsonStrongTypedParsedData)
+        {
+            _inputTextArea.Language(JsonLanguage);
+            _outputTextArea.Language(JsonLanguage);
+            _inputTextArea.Text(jsonStrongTypedParsedData);
+        }
+    }
+
+    public void Dispose()
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _semaphore.Dispose();
+    }
+
+    private void OnSettingChanged(bool value)
+    {
+        StartFormat(_inputTextArea.Text);
+    }
+
+    private void OnIndentationModelChanged(Indentation indentationMode)
+    {
+        StartFormat(_inputTextArea.Text);
+    }
+
+    private void OnInputTextChanged(string text)
+    {
+        StartFormat(text);
+    }
+
+    private void StartFormat(string text)
+    {
+        _cancellationTokenSource?.Cancel();
+        _cancellationTokenSource?.Dispose();
+        _cancellationTokenSource = new CancellationTokenSource();
+
+        WorkTask = FormatAsync(text, _cancellationTokenSource.Token, indentationMode, sortProperties);
+    }
+
+    private async Task FormatAsync(string input, CancellationToken cancellationToken, SettingDefinition<Indentation> indentationSetting, SettingDefinition<bool> sortPropertiesSetting)
+    {
+        using (await _semaphore.WaitAsync(cancellationToken))
+        {
+            await TaskSchedulerAwaiter.SwitchOffMainThreadAsync(cancellationToken);
+
+            ResultInfo<string> formatResult = await JsonHelper.Format(
+                input,
+                _settingsProvider.GetSetting(indentationSetting),
```

Comment:
It would be better to pass the value of the setting in parameter of `FormatAsync` instead of the setting definition. The reason why is what if the setting value changed between the moment when `StartFormat` is invoked and when we reach this line?

Reply:
Same thing in XML Formatter, my bad, sorry for not catching it the last time 😅

Reply:
Fixed in the XML and JSON converters, as well as several other tools.